### ISSUE TITLE
Auto-complete for bin field in package.json confusing for scoped packages

### ISSUE THREAD

**Issue by rockbot** [npm/npm#6658]
If a package is scoped, the package.json looks something like this: 

``` js
{
  "name": "@foo/bar",
  //...
  "bin": "index.js"
  //...
}
```

But when installing that package globally, it's uncertain where the binary is located. The workaround would be to modify the package.json like so:

``` js
{
  "name": "@foo/bar",
  //...
  "bin": {"bar": "index.js"}
  //...
}
```

but it'd be nice if it did the right thing automagically.

:-)

**Comment by watilde** [comment]
I think this feature has already worked well:
> the mapping from the package name to the script name

e.g. https://github.com/npm/npm/blob/master/package.json#L32

**Comment by destroyerofbuilds** [comment]
@othiym23, I've just encountered this myself while setting up a Yeoman generator.

I bootstrap scoped packages, but rely on the behavior of [bin](https://docs.npmjs.com/files/package.json#bin) to use the package name as the name of the binary.

To test I ran `npm link` and here is the output (scope name changed):

```
/home/user/hbetts/.local/var/nvm/versions/node/v0.12.0/bin/@myco/bit -> /home/user/hbetts/.local/var/nvm/versions/node/v0.12.0/lib/node_modules/@myco/bit/bin/cli.js
```

`which bit` fails to find the `bit` executable.
